cub/cub/device/dispatch/dispatch_scan.cuh

```diff
@@ -286,12 +337,13 @@ struct DispatchScan
       int scan_sm_occupancy;
       error = CubDebug(MaxSmOccupancy(scan_sm_occupancy, // out
                                       scan_kernel,
-                                      Policy::BLOCK_THREADS));
+                                      policy.Scan().BlockThreads()));
       if (cudaSuccess != error)
       {
         break;
       }
 
+      // TODO(ashwin): should this come from the launcher factory instead?
       // Get max x-dimension of grid
       int max_dim_x;
       error = CubDebug(cudaDeviceGetAttribute(&max_dim_x, cudaDevAttrMaxGridDimX, device_ordinal));
```

Comment:
suggestion: yes, all CUDA Runtime calls should be consolidated in launcher factory

Reply:
Fixed - thanks. Sorry for missing this TODO.